test/vm-install

```diff
@@ -81,31 +67,48 @@ def build_and_maybe_install(image, do_install=False, skip=None):
         finally:
             machine.stop()
 
-def only_install(image, skip=None):
+def only_install(image, skip=None, args=None):
     """Install Cockpit into a test image"""
     machine = testvm.VirtMachine(verbose=args.verbose, image=image, label="install")
     machine.start(maintain=True)
     try:
         machine.wait_boot()
-        upload_scripts(machine)
+        upload_scripts(machine,args=args)
         machine.execute("rm -rf /var/tmp/build-results");
         machine.upload([ "tmp/build-results" ], "/var/tmp")
-        run_install_script(machine, False, True, skip, None)
+        run_install_script(machine, False, True, skip, None, args)
     finally:
         machine.stop()
 
-try:
-    skip = "cockpit-ostree"
-    if install_image and "fedora-atomic" in install_image:
-        skip = None
+def build_and_install(install_image, build_image, args):
+    try:
+        skip = "cockpit-ostree"
+        if install_image and "fedora-atomic" in install_image:
+            skip = None
+
+        if build_image and build_image == install_image:
+            build_and_maybe_install(build_image, do_install=True, skip=skip, args=args)
+        else:
+            if build_image:
+                build_and_maybe_install(build_image, do_install=False, skip=skip, args=args)
+            if install_image:
+                only_install(install_image, skip, args=args)
+    except testvm.Failure, ex:
+        print >> sys.stderr, "vm-install:", ex
+        sys.exit(1)
```

Comment:
Don't call `sys.exit` from a library function.  It happens to work because build_and_install is the last thing to be called in testsuite_prepare, but this will break too easily.